frontend/src/app/components/network-graph/network-graph.component.html

```diff
@@ -20,11 +20,11 @@
         />
         <button
           mat-button
-          *ngIf="filterControl.value"
           matSuffix
           mat-icon-button
           aria-label="Clear"
           (click)="clearFilter()"
+          [style.visibility]="filterControl.value ? 'visible' : 'hidden'"
```

Comment:
What is the advantage of keeping it in the DOM but hidden instead of just "removing" the entry? So what is the background here? Would it harm the layout or...?

Reply:
Yes exactly, the layout is the problem. The input field grew when the button was shown, but it should keep it size of course. There might be some better approaches with `display: absolute` to remove it from the DOM, but in this case using the visibility was the easiest and I think it is fine.